File: 1275 Find Winner on a Tic Tac Toe Game.java
class Solution {
    public String tictactoe(int[][] moves) {
        int[] cols = new int[3], rows = new int[3], diags = new int[2];
        int val = 1, cnt = 0;
        for (int[] move : moves) {
            int i = move[0], j = move[1];
            rows[i]+= val; cols[j] += val;
            if (i == j) diags[0] += val;
            if (i + j == 2) diags[1] += val;
            
            if (rows[i] == 3 || rows[i] == -3) return rows[i] == 3 ? "A" : "B";
            else if (cols[j] == 3 || cols[j] == -3) return cols[j] == 3 ? "A" : "B";
            else if (diags[0] == 3 || diags[0] == -3 || diags[1] == 3 || diags[1] == -3)
                return (diags[0] == 3 || diags[1] == 3) ? "A" : "B";
            val = -val;
            ++cnt;
        }
        
        return cnt == 9 ? "Draw" : "Pending";        
    }
}

File: 1275 Find Winner on a Tic Tac Toe Game.py
class Solution:
    def __init__(self):
        self.slash = [[0,0],[1,1],[2,2]]
        self.bslash = [[0,2],[1,1],[2,0]]
        self.pl = ["B", "A"]
    def tictactoe(self, moves: List[List[int]]) -> str:
        grid = [[0]*3 for _ in range(3)] # A => 2, B => 1, empty => 0
        cnt = 0
        for move in moves:
            cnt += 1
            grid[move[0]][move[1]] = cnt % 2 + 1

        if self.check(grid, move, cnt):
            return self.pl[cnt % 2]
        elif cnt == 9:
            return "Draw"
        else:
            return "Pending"

    def check(self, g: List[List[int]], m: List[int], cnt: int) -> bool:
        player = cnt % 2 + 1
        r, c = m[0], m[1]
        if ((g[r][0] == g[r][1] == g[r][2]) or 
            (g[0][c] == g[1][c] == g[2][c]) or 
            (m in self.slash and g[0][0] == g[1][1] == g[2][2]) or 
            (m in self.bslash and g[0][2] == g[1][1] == g[2][0])):
            return True
        else:
            return False
        